distributed/distributed-schema.yaml

```
@@ -643,15 +643,20 @@ properties:
             description: |
               Environment variables to set on all worker processes started by nannies.
               These variables are set in the worker process after it has started.
+              To unset a variable in a config override, set its value to null.
+              To unset from DASK_* environment variable, it must be set to None (case
+              sensitive).
 
           pre-spawn-environ:
             type: object
             description: |
               Environment variables to set on all worker processes started by nannies.
-              These variables are set within the Nanny process,
-              before spawning the worker process.
-              Should be used for variables that must be set before
+              These variables are set within the Nanny process, before spawning the
+              worker process. Should be used for variables that must be set before
               process startup, interpreter startup, or imports.
+              To unset a variable in a config override, set its value to null.
+              To unset from a DASK_* environment variable, it must be set to None (case
```

`None` is a Python object, so I'm not following. Are you saying the environment variable needs to be set to `"None"`? (Does Dask convert `DASK_FOO=None` to `{"foo": None}` in Dask config?)

---

Correct. It's because of these lines:
https://github.com/dask/dask/blob/7693c804a3412be26d79dee94d4ebce36d3161e9/dask/config.py#L257-L263

Admittedly this works fine for strings and numbers but it's a bit convoluted for nones, so it could benefit from a patch in dask/dask.

---

```suggestion
              To unset from a DASK_* environment variable, it must be set to "None" (case
```

This would make me less confused.

---

I've tweaked the note

---

done.